bot/exts/code_jams/_views.py

```diff
@@ -134,3 +135,118 @@ async def interaction_check(self, interaction: discord.Interaction) -> bool:
+                log.error(f"Something went wrong: {err}")
+            return
+        else:
+            jam_id = user["team"]["jam_id"]
+            try:
+                await self.mgmt_client.post(
+                    "infractions",
+                    json={
+                        "user_id": self.member.id,
+                        "jam_id": jam_id, "reason": self.note.value,
+                        "infraction_type": "note"
+                    },
+                    raise_for_status=True
+                )
+            except ResponseCodeError as err:
+                if err.response.status == 404:
+                    await interaction.response.send_message(
+                        "Something went wrong! The user could not be found!",
+                        ephemeral=True
+                    )
+                else:
+                    await interaction.response.send_message(":x: Something went wrong, we have notified the team!")
+                    log.error(f"Something went wrong: {err}")
+                return
+            else:
+                await interaction.response.send_message('Your note has been saved!', ephemeral=True)
+
+    async def on_error(self, error: Exception, interaction: discord.Interaction) -> None:
+        """Discord.py default to handle modal error."""
+        await interaction.response.send_message('Something went wrong with processing your form', ephemeral=True)
+
+
+class JamInfoView(discord.ui.View):
+    """
+    A basic view that displays basic information about a CJ participant.
+
+    Additionally, you can either view the notes that was added to the participant
+    or create one of them.
+    """
+
+    def __init__(self, member: discord.Member, mgmt_client: APIClient):
+        super().__init__(timeout=900)
+        self.mgmt_client = mgmt_client
+        self.member = member
+
+    @discord.ui.button(label='Add Note', style=discord.ButtonStyle.green)
+    async def add_note(self, button: discord.ui.Button, interaction: discord.Interaction) -> None:
+        """A button to add a note."""
+        await interaction.response.send_modal(AddNoteModal(self.member, self.mgmt_client))
+
+    @discord.ui.button(label='View notes', style=discord.ButtonStyle.green)
+    async def view_notes(self, button: discord.ui.Button, interaction: discord.Interaction) -> None:
+        """A button to view the notes of a participant."""
+        try:
+            user = await self.mgmt_client.get(f"users/{quote_url(str(self.member.id))}", raise_for_status=True)
+        except ResponseCodeError as err:
+            if err.response.status == 404:
+                await interaction.response.send_message(":x: Something went wrong, the user could not be found!")
+            else:
+                await interaction.response.send_message(":x: Something went wrong, we have notified the team!")
+                log.error(f"Something went wrong: {err}")
+            return
```

Comment:
This appears to be a duplicate of what's in `AddNoteModal`. Would be worthwhile to create a function. In any case, same comments/suggestions apply to this code.

Reply:
Actually, it uses a slightly different endpoint. As for viewing a note, we need the participation history, which contains infraction related data from all the previous jams, whereas the endpoint what we call in `on_submit` is another endpoint that returns data regarding the current jam, as we're going to add infraction in regards of the **current jam.**

Reply:
If the endpoints are different, the endpoint could be an argument for the new function this is moved to. The error handling is the same and that's the bulk of the code.